moped-toolbox/amd_milestones_backfill/create_milestones.py

```diff
@@ -0,0 +1,104 @@
+"""The script inserts moped_proj_milestones into signal and PHB projects. It was created
+to backfill project milestones after implementing the milestone template feature for
+AMD: https://github.com/cityofaustin/atd-data-tech/issues/9102.
+
+Usage:
+    python create_milestones.py -e local -d 2023-01-01
+"""
+import argparse
+
+from utils import get_logger, make_hasura_request, TEMPLATE_MILESTONES
+from queries import PROJECTS_QUERY, PROJ_MILESTONES_MUTATION
+from secrets import HASURA
+
+
+def main(env, max_date_added):
+
+    logger.info(f"Gettings projects added on or before {max_date_added} in {env} env")
+
+    data = make_hasura_request(
+        query=PROJECTS_QUERY,
+        variables={"max_date_added": max_date_added},
+        endpoint=HASURA["hasura_graphql_endpoint"][env],
+        admin_secret=HASURA["hasura_graphql_admin_secret"][env],
+    )
+
+    projects = data["moped_project"]
+    all_milestones = data["moped_proj_milestones"]
+
+    logger.info(f"{len(projects)} projects to process")
+
+    update_count = 0
+
+    for project in projects:
+        project_id = project["project_id"]
+
+        existing_milestone_ids = [
+            m["moped_milestone"]["milestone_id"]
+            for m in all_milestones
+            if m["project_id"] == project_id
+        ]
+
+        proj_milestones_new = []
+
+        for m in TEMPLATE_MILESTONES:
+            if m["milestone_id"] not in existing_milestone_ids:
```

Comment:
this script can be run repeatedly with same args without concern for duplicating milestones. if a project already has a milestone ID specified in the template, that milestone is skipped.

the script is making the assumption that the `moped_milestone` IDs in local, staging, and project all match (they do)